test/integration/component/test_ospf.py

```diff
@@ -0,0 +1,477 @@
+        
+        self.vpc_off = VpcOffering.create(
+                                     self.apiclient,
+                                     self.services["vpc_offering"]
+                                     )
+        self.vpc_off.update(self.apiclient, state='Enabled')
+        
+        self.validate_vpc_offering(self.vpc_off)
+        self.logger.debug("VPC Offering id = %s" % self.vpc_off.id)
+        
+        """  vpc_super_netmask=255.255.252.0 """
+        self.batman_vpc = VPC.create(
+            self.apiclient,
+            self.services["batman_vpc"],
+            netmask="255.255.252.0",
+            vpcofferingid=self.vpc_off.id,
+            zoneid=self.zone.id
+        )
+        
+        self.validate_vpc_network(self.batman_vpc)
+        self.logger.debug("VPC batman_vpc id = %s" % self.batman_vpc.id)
+        
+        self.superman_vpc = VPC.create(
+            self.apiclient,
+            self.services["superman_vpc"],
+            netmask="255.255.252.0",
+            vpcofferingid=self.vpc_off.id,
+            zoneid=self.zone.id
+        )
+        
+        self.validate_vpc_network(self.superman_vpc)
+        self.logger.debug("VPC superman_vpc id = %s" % self.superman_vpc.id)
+        
+        """ -----------------------------------------------"""
+        
+        self.network_offering = NetworkOffering.create(
+            self.apiclient,
+            self.services["network_offering"],
+            conservemode=False
+        )
+        # Enable Network offering
+        self.network_offering.update(self.apiclient, state='Enabled')
+        self.logger.debug("Dynamic Network offering id = %s" % self.network_offering.id)
+        
+        """ --------------------BATMAN---------------------------"""
+          # 2) Create ACL
+        self.batman_aclgroup = NetworkACLList.create(apiclient=self.apiclient, services={}, name="acl", description="acl", vpcid=self.batman_vpc.id)
+        self.assertIsNotNone(self.batman_aclgroup, "Failed to create NetworkACL list")
+        self.logger.debug("Created a network ACL list %s" % self.batman_aclgroup.name)
+
+        # 3) Create ACL Items
+        self.batman_aclitem1 = NetworkACL.create(apiclient=self.apiclient, services={},
+            protocol="ICMP", action="Allow", aclid=self.batman_aclgroup.id, cidrlist=["0.0.0.0/0"])
+        self.assertIsNotNone(self.batman_aclitem1, "Network failed to aclItem")
+        self.logger.debug("Added a network ACL %s to ACL list %s" % (self.batman_aclitem1.id, self.batman_aclgroup.name))
+        
+        self.batman_aclitem2 =  NetworkACL.create( self.apiclient, aclid=self.batman_aclgroup.id, services=self.services["natrule"], 
+                                                   traffictype='Ingress' )
+        
+        self.assertIsNotNone(self.batman_aclitem2, "Network failed to aclItem")
+        self.logger.debug("Added a network ACL %s to ACL list %s" % (self.batman_aclitem2.id, self.batman_aclgroup.name))
```

Comment:
Consider extracting lines 315-325 to a private method to reduce the complexity of this test method.